src/test/java/io/xstefank/wildfly/bot/PRAutomaticLabelingTest.java

```diff
@@ -0,0 +1,88 @@
+    private static PullRequestJson pullRequestJson;
+    private MockedContext mockedContext;
+
+    @BeforeAll
+    static void setupTests() throws IOException {
+        pullRequestJson = PullRequestJson.builder(TestConstants.VALID_PR_TEMPLATE_JSON)
+                .action(SYNCHRONIZE)
+                .build();
+    }
+
+    @Test
+    public void testNotApplyAndRemoveRebaseThisLabel() throws IOException {
+        mockedContext = MockedContext.builder(pullRequestJson.id())
+                .prLabels(LABEL_NEEDS_REBASE, LABEL_FIX_ME);
+        wildflyConfigFile = """
+                wildfly:
+                  rules:
+                    - id: "Label rule"
+                      title: WFLY
+                       """;
+        given().github(mocks -> Util.mockRepo(mocks, wildflyConfigFile, pullRequestJson, mockedContext))
+                .when().payloadFromString(pullRequestJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    Mockito.verify(mocks.pullRequest(pullRequestJson.id()), Mockito.never())
+                            .addLabels(ArgumentMatchers.anyString());
+                    final ArgumentCaptor<String[]> argumentCaptor = ArgumentCaptor.forClass(String[].class);
+                    Mockito.verify(mocks.pullRequest(pullRequestJson.id())).removeLabels(argumentCaptor.capture());
+                    MatcherAssert.assertThat(Arrays.asList(argumentCaptor.getValue()),
+                            Matchers.containsInAnyOrder(LABEL_NEEDS_REBASE, LABEL_FIX_ME));
+                });
+    }
+
+    @Test
+    public void testApplyRebaseThisLabel() throws IOException {
+        wildflyConfigFile = """
+                wildfly:
+                  rules:
+                    - id: "Label rule"
+                      title: WFLY
+                       """;
+        mockedContext = MockedContext.builder(pullRequestJson.id())
+                .mergeable(Boolean.FALSE);
+        given().github(mocks -> Util.mockRepo(mocks, wildflyConfigFile, pullRequestJson, mockedContext))
+                .when().payloadFromString(pullRequestJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    ArgumentCaptor<String[]> argumentCaptor = ArgumentCaptor.forClass(String[].class);
+                    Mockito.verify(mocks.pullRequest(pullRequestJson.id())).addLabels(argumentCaptor.capture());
+                    MatcherAssert.assertThat(Arrays.asList(argumentCaptor.getValue()),
+                            Matchers.containsInAnyOrder(LABEL_NEEDS_REBASE));
+                    // we need to capture it otherwise we would not be able to correctly verify this
+                    //  any() fails on ambigious call, which is only matcher for vararg params
+                    Mockito.verify(mocks.pullRequest(pullRequestJson.id()), Mockito.never())
+                            .removeLabels(argumentCaptor.capture());
```

Comment:
verify capture?

Reply:
I have also commented it there. We are not able to otherwise verify, that this method was/wasn't invoked. for example `any()` or `anyCollection()` matcher would not work